cli/flox-rust-sdk/src/models/pkgdb.rs

```diff
@@ -114,6 +116,11 @@ impl<'de> Deserialize<'de> for PkgDbError {
             message: m,
             caught: caught_message_contents.map(|m| CaughtMsgError { message: m }),
         });
+
+        if let Some(k) = map.keys().next() {
```

Comment:
Suggestion: I don't think we should do this. At this point we're already in the process of generating an error message for the user and we've already done the work of processing it, but if we get an extra field we're going to throw away the actual error and instead raise an error about there being an unexpected field in the JSON.

Reply:
yeah it's odd
and makes it harder to change the error object without _immidiately_ changing the rust as well.
logging this might be good enough if not ignoring it altogether

Reply:
I changed it because it was causing us to print:
```
[ERROR] [flox] error building environment:
```
as in https://github.com/flox/flox/issues/658.

In the C++, we were still using the old JSON schema (`{message: ..., error_code: ...}`) in one place, and rather than detecting that, we just had bad behavior. So yes, this would mean you have to immediately change the Rust schema if you change the C++ schema, but I see that as better than overlooking a change in schema.

Reply:
I'm not really concerned about how hard it is to change the error messages, but I really think it's bad for the user to drop an existing error because the JSON contained extra data. If we're concerned about JSON-formatted errors containing extra data we could add debug assertions so that they're caught at development time.

Reply:
It's more about protecting against loss of information than dropping extra data. If e.g. we rename `context_message` to `context-message` in Rust and forget to change that in C++, we wouldn't necessarily notice, we'd just never have a context message.

I can make it a debug assertion